ruby-challenges.rb

```diff
@@ -10,18 +10,93 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# ## Define a new method called word_checker
+# ## This method will have 2 parameters, one for the array of strings called array, and one called 'letter' for the variables storing strings
+# ## define a new varialbe inside the method called new_array that will store the array with .select attached. 
+# ## .select works similarly to .filter in JavaScript
+# ## call the value string to help the flow of the code.
+# ## create an If statement that will act as the logic for the .select.
+# ## using the string value with .include? attached to check for the letter parameter.
+# ## return the string 
+# ## add an else statement with returns nil 
+# ## Before the last end do a final return of new_array to display all the strings needed. e 
 
-# -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
-# HINT: Google 'ruby get rid of nested arrays'
+def word_checker array, letter
+  new_array = array.select do |string|
+    if string.include?(letter)
+     string
+    else
+      nil
+    end
+  end
+  return new_array
+end
+
```

Comment:
Nice work! A couple notes:
1) `.select` is similar to `.filter` in JavaScript, as it has a built in `if` statement. It will only "select" items that agree with whatever condition you have made. This means there is no need for an `if` statement.
2) Whenever you create a variable just to return it right after, you can always just skip creating that variable and return that data instead!